pubsublite/publish/publish_batch.go

```diff
@@ -63,19 +63,24 @@ func publishWithBatchSettings(w io.Writer, projectID, zone, topicID string, mess
 	}
 
 	// Print publish results.
+	var publishedCount int
 	for _, r := range results {
 		// Get blocks until the result is ready.
 		id, err := r.Get(ctx)
 		if err != nil {
-			// NOTE: The publisher will terminate upon first error. Create a new
-			// publisher to republish failed messages.
-			return fmt.Errorf("publish error: %v", err)
+			// NOTE: A failed PublishResult indicates that the publisher client
+			// encountered a fatal error and has permanently terminated. After the
+			// fatal error has been resolved, a new publisher client instance must be
+			// created to republish failed messages.
```

Comment:
Should we show how to collect the failed messages that need to be resent?

Reply:
Good idea. I updated the quickstart_publisher sample to demonstrate this. But left the other snippets more minimal.